src/main/java/spoon/reflect/ModelStreamer.java

```diff
@@ -35,20 +35,24 @@
 	 * 		the factory to be save
 	 * @param out
 	 * 		the used output stream
+	 * @param zipIt
+	 * 		true if output file has to be compressed
 	 * @throws IOException
 	 * 		if some IO error occurs
 	 */
-	void save(Factory f, OutputStream out) throws IOException;
+	void save(Factory f, OutputStream out, boolean zipIt) throws IOException;
```

Comment:
I wonder if it wouldn't be interesting to propose the compression by default.
Then to keep the old signature of those methods, and to add a boolean `zipIt` and a new setter in the API for deactivating it.

WDYT?

Reply:
Good idea! I'll try to do it tonight.

Reply:
Excellent! In Spoon, we care a lot about backward compatibility, so we generally prefer to add new methods than changing the signature of existing ones.

Reply:
I propose to use an enum rather than a boolean. This allows to add further compression types (e.g. lzma) in future.